software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,640 @@
+#
+#   vanilla_stats_parser.py
+#
+#   vanilla core stats extractor
+# 
+#   input: vanilla_stats.log
+#   output: manycore_stats.log
+#   output: tile_stats/tile_#_stats.log for all tiles 
+#
+#   @author Borna
+#
+#   How to use:
+#   python3 generate_stats.py --dim-y {manycore_dim_y}  --dim-x {manycore_dim_x} 
+#                             --tile (optional) --tile_group (optional)
+#                             --input {vanilla_stats.log}
+#
+#   ex) python3 --input generate_stats.py --dim-y 4 --dim-x 4 --tile --tile_group --input vanilla_stats.log
+#
+#   {manycore_dim_y}  Mesh Y dimension of manycore default = 4
+#   {manycore_dim_x}  Mesh X dimension of manycore default = 4
+#   {per_tile}        Generate separate stats file for each tile default = False
+#   {input}           Vanilla stats input file     default = vanilla_stats.log
+
+
+
+import sys
+import argparse
+import os
+import re
+import csv
+from enum import Enum
+from collections import Counter
+
+
+
+
+
+
+# These values are used by the manycore library in bsg_print_stat instructions
+# they are added to the tag value to determine the tile group that triggered the stat
+# and also the type of stat (stand-alone stat, start, or end)
+# the value of these paramters should match their counterpart inside 
+# bsg_manycore/software/bsg_manycore_lib/bsg_manycore.h
+# Formatting for bsg_cuda_print_stat instructions
+# < stat_type >
+# Section                 Stat type  -   y cord   -   x cord   -    tile group id   -        tag
+# of bits                <----2----> -   <--6-->  -   <--6-->  -   <------10----->  -   <-----8----->
+# Stat type value: {"stat":0, "start":1, "end":2}
+bsg_STAT_TAG_BITS   = 8
+bsg_STAT_TG_ID_BITS = 10
+bsg_STAT_X_BITS     = 6
+bsg_STAT_Y_BITS     = 6
+bsg_STAT_TYPE_BITS  = 2
+bsg_STAT_TAG_MASK   = ((1 << bsg_STAT_TAG_BITS) - 1)
+bsg_STAT_TG_ID_MASK = ((1 << bsg_STAT_TG_ID_BITS) - 1)
+bsg_STAT_X_MASK     = ((1 << bsg_STAT_X_BITS) - 1)
+bsg_STAT_Y_MASK     = ((1 << bsg_STAT_Y_BITS) - 1)
+bsg_STAT_TYPE_STAT  = 0
+bsg_STAT_TYPE_START = 1
+bsg_STAT_TYPE_END   = 2
+
+
+bsg_TILE_GROUP_ORG_X = 0
```

Comment:
Are these still used?

Reply:
No, removed, thanks for pointing it out!